shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -121,13 +123,15 @@ public void writeFields(DataOutput out) throws IOException {
             out.writeLong(requestId);
             out.writeUTF(segment);
             out.writeUTF(correctHost);
+            out.writeUTF(serverStackTrace);
         }
 
         public static WireCommand readFrom(DataInput in, int length) throws IOException {
             long requestId = in.readLong();
             String segment = in.readUTF();
             String correctHost = in.readUTF();
-            return new WrongHost(requestId, segment, correctHost);
+            String serverStackTrace = (((ByteBufInputStream) in).available() > 0) ? in.readUTF() : EMPTY_STACK_TRACE;
```

Comment:
The interface this implements actually has a ByteBufInputStream in the interface. So you can change the type of in in the signature on line 129 to ByteBufInputStream without having to change any other code.